dev/managedservicesplatform/internal/resource/sentryalert/sentryalert.go

```diff
@@ -0,0 +1,241 @@
+package sentryalert
+
+import (
+	"encoding/json"
+
+	"github.com/aws/constructs-go/constructs/v10"
+
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+
+	sentryproject "github.com/sourcegraph/managed-services-platform-cdktf/gen/sentry/project"
+
+	"github.com/sourcegraph/managed-services-platform-cdktf/gen/sentry/issuealert"
+
+	"github.com/sourcegraph/sourcegraph/dev/managedservicesplatform/internal/resourceid"
+	"github.com/sourcegraph/sourcegraph/lib/pointers"
+)
+
+type Output struct{}
+
+// Config for a Sentry issue alert.
+//
+// [Sentry] and [Terraform] document each field
+//
+// [Sentry]: https://docs.sentry.io/api/alerts/create-an-issue-alert-rule-for-a-project/
+// [Terraform]: https://registry.terraform.io/providers/jianyuan/sentry/latest/docs/resources/issue_alert
+type Config struct {
+	// SentryProject is the project to set the alert on
+	SentryProject sentryproject.Project
+	// AlertConfig is the configuration for the Sentry issue alert rule
+	AlertConfig AlertConfig
+}
+
+// AlertConfig is the configuration for the Sentry issue alert rule
+type AlertConfig struct {
+	// Id of the issue alert. Must be unique
+	Id string
+	// Name the name of the alert
+	Name string
+	// Frequency determines how often to perform the actions for an issue, in minutes. (valid range 5-43200)
+	Frequency float64
+	// ActionMatch trigger actions when an event is captured by Sentry and `ActionMatch` of the
+	// specified `Conditions` happen
+	ActionMatch ActionMatch
+	// Conditions which must be satisfied for actions to trigger.
+	// `ActionMatch` determines whether all or any must be true.
+	Conditions []Condition
+	// Actions the list of actions to run when the conditions match
+	Actions []Action
+
+	// FilterMatch (optional) determines which filters need to be true before actions are executed.
+	// Required when a value is provided for `Filters`
+	FilterMatch FilterMatch
+	// Filters determines if a rule fires after the necessary conditions have been met
+	Filters []Filter
+}
+
+func (a AlertConfig) Validate() error {
+	if a.Name == "" {
+		return errors.New("Name is required")
+	}
+
+	if a.ActionMatch == "" {
+		return errors.New("ActionMatch is required")
+	}
+
+	// TODO(jac): allow Conditions to be nil after provider issue is fixed
+	// https://github.com/jianyuan/terraform-provider-sentry/issues/366
+	if a.Conditions == nil || len(a.Conditions) == 0 {
```

Comment:
FYI this works the same way and won't panic

```suggestion
	if len(a.Conditions) == 0 {
```

Same for other slice length checks, if you want to update.